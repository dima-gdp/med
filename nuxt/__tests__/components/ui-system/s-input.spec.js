import { mount } from '@vue/test-utils'
import SInput from '~/components/ui-system/s-input'

describe('ui-system/s-input', () => {
  const INPUT_DEFAULT_PROPS = {
    value: '',
    placeholder: '',
    disabled: false,
    appendIcon: '',
    type: 'text',
    autocomplete: 'off',
    name: '',
    readonly: false,
  }

  const inputFactory = (propsData = {}) => {
    return mount(SInput, {
      propsData: {
        ...propsData,
      },
      attachTo: document.body,
    })
  }

  it('s-input отображается', () => {
    const wrapper = inputFactory(INPUT_DEFAULT_PROPS)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('s-input дефолтные значения', () => {
    const wrapper = inputFactory(INPUT_DEFAULT_PROPS)
    const input = wrapper.find('input')

    expect(wrapper.classes('s-input')).toBe(true)
    expect(input.attributes('autocomplete')).toBe('off')
    expect(input.attributes('type')).toBe('text')
    expect(input.attributes('name')).toBe('')
    expect(input.attributes('placeholder')).toBe('')
  })

  it('s-input при установленном значении disabled добавляется class (s-input--disabled) и атрибут disabled', () => {
    const wrapper = inputFactory({ value: '', disabled: true })
    const input = wrapper.find('input')

    expect(wrapper.classes('s-input--disabled')).toBe(true)
    expect(input.attributes('disabled')).toBeTruthy()
  })

  it('s-input при вводе значения эммитит событие input с введенымм значением', async () => {
    const wrapper = inputFactory(INPUT_DEFAULT_PROPS)
    const input = wrapper.find('input')
    await input.trigger('input')

    expect(wrapper.emitted().input).toEqual([['']])
  })

  it('s-input при focus эммитит событие focus со значением true ', () => {
    const wrapper = inputFactory(INPUT_DEFAULT_PROPS)
    const input = wrapper.find('input')
    input.trigger('focus')
    expect(wrapper.emitted().focus).toEqual([[true]])
  })

  it('s-input при blur эммитит событие blur со значением false', () => {
    const wrapper = inputFactory(INPUT_DEFAULT_PROPS)
    const input = wrapper.find('input')
    input.trigger('blur')
    expect(wrapper.emitted().blur).toEqual([[false]])
  })
})
