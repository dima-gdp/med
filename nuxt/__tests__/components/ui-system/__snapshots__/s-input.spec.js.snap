// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-system/s-input s-input отображается 1`] = `
<div class="s-input"><input autocomplete="off" type="text" name="" placeholder="" class="s-input__input t-p">
  <!---->
</div>
`;
